feelslikemonday/app/src/main/java/com/example/feelslikemonday/DAO/UserDAO.java

```diff
@@ -68,4 +78,25 @@ public void onFailure(@NonNull Exception e) {
                     }
                 });
     }
+
+    /**
+     * Queries for a user by the user's username. A callback method will be called on success
+     */
+    public void get(String username, final UserCallback onSuccess){
+        db.collection("users")
+                .whereEqualTo(username, true)
+                .get()
+                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                        if (task.isSuccessful() && task.getResult() != null) {
+                            // Assumes only one document will be returned
+                            onSuccess.onCallback((User)task.getResult().getDocuments().get(0).getData());
```

Comment:
For the user getter, we're always going to want a user returned to us, so would it make more sense and would it be possible to just implement the callback function in the OnComplete method to return a user object to whatever called the getter?

Reply:
Depends on how things are timed. May or may not work depending on what is executed when